mining effect fix

Lajnux opened the pull request:
### Description:

fixed so that mining gloves and varrock armour use the updated logics.

### Changes:

fixed so that mining gloves and varrock armour use the updated logics.

### Other checks:

- [x] I have tested all my changes thoroughly.

gc commented:
I refactored the mining code so it can print out a table of the xp/hr in the wiki, so that I can see if this PR effects those. Running `yarn wiki` will update this wiki page based on your code, although some merge conflicts will have to be fixed.

![image](https://github.com/user-attachments/assets/40ec82d1-03fa-426a-b384-12f194bbee44)

Lajnux commented:
> I refactored the mining code so it can print out a table of the xp/hr in the wiki, so that I can see if this PR effects those. Running `yarn wiki` will update this wiki page based on your code, although some merge conflicts will have to be fixed.
> 
> ![image](https://private-user-images.githubusercontent.com/30398469/376892911-40ec82d1-03fa-426a-b384-12f194bbee44.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwNTg5MjgsIm5iZiI6MTczMDA1ODYyOCwicGF0aCI6Ii8zMDM5ODQ2OS8zNzY4OTI5MTEtNDBlYzgyZDEtMDNmYS00MjZhLWIzODQtMTJmMTk0YmJlZTQ0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI3VDE5NTAyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRmZmIxZTcwOGMyNmE0ZmFiNGIyNzM4NTRkYjc0ZDNkYWNmNDZkZjQyYWE5YmJkNjQ1NGU5ZTlmMzQ1NjFiMWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.C3oeVM8QWxZfFIywZtolVUZXOOipmc4oKBpwqBBQtDo)

I fixed the merge conflict, not sure if the yarn wiki went good tho, my lint seems slightly different
